w3m-hist.el

```diff
@@ -718,6 +718,116 @@ are accessible by PREV and NEXT operations."
     (setq w3m-history-flat (w3m-history-slimmed-history-flat))
     (w3m-history-tree position)))
 
+(defvar w3m-scrub-command-list
+  '(("/usr/bin/shred" "-fuz") ("/bin/rm" "-f"))
+  "List of methods to use for attempting to securely delete temporary files.
+
+Function `w3m-scrub-command' uses the first entry on the list
+that it finds available on the host OS; if none are found, it
+will use the insecure emacs function `delete-file'.")
+
+(defun w3m-history-scrub ()
+  "Initializes history, cache, cookies, and temp files.
+
+The method used to delete files is configurable to your personal
+level of paranoia, via `w3m-scrub-command-list'.
+
+Note that some operations performed by this function may impact
+operation of the external w3m browser. Also, be advised that
+other emacs features outside the scope of `emacs-w3m' may
+independantly be saving varibles, for example in order to restore
+a crashed emacs session."
+  ; TODO:
+  ; + Use a defcustom or other user-friendly mechanism to allow
+  ;   fine-grained customization of what to scrub.
+  ; + When invoked with a prefix arg, allow the scrub to be limited to a
+  ;   timeframe, for elements that store timestamp information.
+  (interactive)
+  (let* ((w3m-fb-mode nil)
```

Comment:
yes-or-no-p() should be called to proceed the following destructive process.  Or, `force' optional argument should be added and yes-or-no-p() may be called from interactive() form.

Reply:
Agreed. Done, in 0c6f5ea